util/src/main/java/com/redhat/lightblue/util/metrics/DropwizardRequestMetrics.java

```diff
@@ -131,7 +133,9 @@ public void endRequestMonitoring() {
 
         @Override
         public void markRequestException(Error e) {
-            String errorCode = e.getErrorCode().replaceAll(REGEX, ".");
+            // Presence of : in metricnamespace makes it's display a bit weird in visualvm.
+            // This might not effect the metric in graphite, but replacing it as a precaution.
+            String errorCode = e.getErrorCode().replace(":", "-");
```

Comment:
How about replacing this with _ like we do with version? It might be surprising to use a different replacement character I think.

We also still might want to replace a few other things, like any whitespace that might creep into error codes. AFAIK there isn't a well-defined set of valid characters in error codes, so I did like that you were being relatively cautious before.

Reply:
Added Regex pattern matching back to error code. This time it replaces any special character with "_"